### PR TITLE
RUM-462 / RUM-1964 add OTel support

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

Adds OTel support and let user chose their tracing header per host.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -92,7 +92,10 @@ which supports most features of the `roUrlTransfer` component (except anything r
 For example, here's how to do a `GetToString` call:
 
 ```brightscript
-    ddUrlTransfer = datadogroku_DdUrlTransfer(m.global.datadogRumAgent)
+    validHosts = {}
+    validHosts["example.com"] = "tracecontext" ' add tracing to requests to "example.com" urls using W3C's traceparentd header
```
Comment (brett0000FF) [inline README.md:96]:
```suggestion
    validHosts["example.com"] = "tracecontext" ' add tracing for requests to "example.com" URLs using W3C's traceparent header
```

**Review by brett0000FF — DISMISSED** [review]

File: library/source/wrapper/DdUrlTransfer.bs [hunk 2]
```diff
@@ -627,29 +640,141 @@ class DdUrlTransfer
...
+        m.headers.Delete("X-B3-TraceId")
+        m.headers.Delete("X-B3-SpanId")
+        m.headers.Delete("X-B3-Sampled")
+        m.headers.Delete("traceparent")
+        m.headers.Delete("tracestate")
+    end sub
+
+    ' ----------------------------------------------------------------
+    ' (Internal) applies the headers recorded in the ddUrlTransfer
+    ' to the underlyins roUrlTransfer component.
```
Comment (0xnm) [inline library/source/wrapper/DdUrlTransfer.bs:759]:
```suggestion
    ' to the underlying roUrlTransfer component.
```

**Review by 0xnm — APPROVED** [review]
lgtm! the only difference with Android/iOS implementation is `t.usr.id` tag, so if we want to have it on Android/iOS, we need to create a task.

**Review by 0xnm — APPROVED** [review]

File: library/source/wrapper/DdUrlTransfer.bs [hunk 3]
```diff
@@ -627,29 +639,129 @@ class DdUrlTransfer
...
+            m.spanId = generateUniqueId(16)
+            hexSpanId = padLeft(m.spanId, 16, "0")
+            traceparent = "00-" + padLeft(m.traceId, 32, "0") + "-" + hexSpanId + "-01"
+            m.AddHeader("traceparent", traceparent)
+            usrId = m.global.datadogUserInfo.id
+            if (usrId <> invalid)
+                usrIdByteArray = CreateObject("roByteArray")
+                usrIdByteArray.FromAsciiString(usrId)
+                usrIdBase64 = usrIdByteArray.ToBase64String().Replace("=", "~")
+                tracestate = "dd=s:1;o:rum;p:" + hexSpanId + ";t.usr.id:" + usrIdBase64
```
[the remark this answers is not in the document]
Comment (ganeshnj) [reply library/source/wrapper/DdUrlTransfer.bs:706]: tags are usually not supported in the SDKs but if we ever supported them, then yes.

I would suggest, we shouldn't introduce this until someone asks for it.

**Pull request merged**